src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/handlers/VideoBookingAmendedEventHandler.kt

```diff
@@ -45,43 +47,90 @@ class VideoBookingAmendedEventHandler(
+                  }
                 }
               }
+              BookingType.PROBATION -> amendIfNotTheSame(oldBooking.probationMeeting(), amendedBooking.probationMeeting()!!)
             }
           } else {
+            // All cancellations must be done first so as not to interfere with new appointments
             // There is no ability to amend appointments for non-rolled out prisons so we have to cancel and create
-            history.appointments().forEach {
-              manageExternalAppointmentsService.cancelPreviousAppointment(it)
-            }
-
-            booking.appointments().forEach {
-              manageExternalAppointmentsService.createAppointment(it)
-            }
+            oldBooking.appointments().forEach(manageExternalAppointmentsService::cancelPreviousAppointment)
+            amendedBooking.appointments().forEach(manageExternalAppointmentsService::createAppointment)
           }
 
-          log.info("Processed BOOKING_AMENDED event for videoBookingId ${booking.videoBookingId}")
+          log.info("Processed BOOKING_AMENDED event for videoBookingId ${amendedBooking.videoBookingId}")
         },
         {
           // Ignore, there is nothing we can do if we do not find the booking
           log.warn("Video booking with ID ${event.additionalInformation.videoBookingId} not found")
         },
       )
   }
+
+  private fun PrisonAppointment?.isTheSameAs(historyAppointment: BookingHistoryAppointment?) = run {
+    this != null &&
+      videoBooking.videoBookingId == historyAppointment?.bookingHistory?.videoBookingId &&
+      appointmentDate == historyAppointment.appointmentDate &&
+      startTime == historyAppointment.startTime &&
+      endTime == historyAppointment.endTime &&
+      prisonerNumber == historyAppointment.prisonerNumber &&
+      prisonLocationId == historyAppointment.prisonLocationId
+  }
+
+  private fun PrisonAppointment.isEarlierThanBefore(historyAppointment: BookingHistoryAppointment) = run {
+    appointmentDate.atTime(startTime).isBefore(historyAppointment.appointmentDate.atTime(historyAppointment.startTime))
+  }
+
+  private fun cancelIfRemoved(old: BookingHistoryAppointment?, new: PrisonAppointment?) {
+    old?.takeIf { new == null }?.run(manageExternalAppointmentsService::cancelPreviousAppointment)
+  }
+
+  private fun createOrAmend(old: BookingHistoryAppointment?, amended: PrisonAppointment) {
+    if (old == null) {
+      manageExternalAppointmentsService.createAppointment(amended)
+    } else {
+      amendIfNotTheSame(old, amended)
+    }
+  }
+
+  private fun createIfNo(old: BookingHistoryAppointment?, amended: PrisonAppointment) {
+    if (old == null) {
+      manageExternalAppointmentsService.createAppointment(amended)
+    }
+  }
+
+  private fun amendIfNotTheSame(old: BookingHistoryAppointment, amended: PrisonAppointment) {
+    if (amended.isTheSameAs(old)) {
+      log.info("No change to meeting date and times, ignoring amend for appointment $amended")
```

Comment:
Took the view we can potentially avoid an API call if there is no actual difference.